app/CMakeLists.txt

```
@@ -350,8 +350,31 @@ if(CELERITAS_BUILD_DEMOS AND CELERITAS_USE_Geant4)
     demo-geant-integration/SensitiveHit.cc
     demo-geant-integration/TrackingAction.cc
   )
+  set(_demo_geant_integration_libs Celeritas::accel)
+
+  if(CELERITAS_USE_ROOT)
+    list(APPEND _demo_geant_integration_src
+      demo-geant-integration/HitRootIO.cc
+    )
+    list(APPEND _demo_geant_integration_libs ROOT::Tree HitClassesDict)
+    include_directories(
+      ${CMAKE_CURRENT_SOURCE_DIR}/demo-geant-integration
+      ${ROOT_INCLUDE_DIRS} ${Geant4_INCLUDE_DIR}
+    )
+    REFLEX_GENERATE_DICTIONARY(HitClasses
```

`ROOT` has 2 syntax to request dictionary `LinkDef` and `xml`.  We should explicitly decide whether the `Celeritas` build will use one or the other or both (with this PR it will use both).

---

Good to know.

---

@sethrj Do you have any opinion on LinkDef vs xml vs both?

---

I don't have a preference; sorry our comments overlapped in time so this is a somewhat duplicated discussion. Is there a history behind the two (is one newer)? Does one have an advantage besides "easier to type" vs "easier to computer generate"?

---

The xml version is newer (came via a merge from an alternate implementation of the dictionary).  LinkDef is more compact.  xml version has a grammar and has one additional feature (ability to add IO annotation of the data members in the xml files instead of the header files; this is necessary for third party header, dubious for own headers as it separates declaration and I/O annotation).

---

I vote for XML then. Maybe you or @stognini could update the existing linkdef files?

One superficial change here is to use lowercase for cmake function calls:
```suggestion
    reflex_generate_dictionary(HitClasses
```

---

Changed.

---

> I vote for XML then. Maybe you or @stognini could update the existing linkdef files?

I will update/change that code.